Source/Charts/Data/Implementations/Standard/ChartData.swift

```diff
@@ -757,3 +764,132 @@ open class ChartData: NSObject
         return max
     }
 }
+
+// MARK: MutableCollection
+extension ChartData: MutableCollection
+{
+    public typealias Index = Int
+    public typealias Element = ChartDataSetProtocol
+
+    public var startIndex: Index
+    {
+        return _dataSets.startIndex
+    }
+
+    public var endIndex: Index
+    {
+        return _dataSets.endIndex
+    }
+
+    public func index(after: Index) -> Index
+    {
+        return _dataSets.index(after: after)
+    }
+
+    public subscript(position: Index) -> Element
+    {
+        get{ return _dataSets[position] }
+        set{ self._dataSets[position] = newValue }
+    }
+}
+
+// MARK: RandomAccessCollection
+extension ChartData: RandomAccessCollection
+{
+    public func index(before: Index) -> Index
+    {
+        return _dataSets.index(before: before)
+    }
+}
+
+// MARK: RangeReplaceableCollection
+extension ChartData: RangeReplaceableCollection
+{
+    public func append(_ newElement: Element)
+    {
+        _dataSets.append(newElement)
+        calcMinMax(dataSet: newElement)
+    }
+
+    public func remove(at position: Index) -> Element
+    {
+        let element = _dataSets.remove(at: position)
+        calcMinMax()
```

Comment:
shall we use `notifyDataChanged` instead of `calcMinMax` here? in CombinedChartData, they works differently then their super class.

Reply:
I thought about that. I think we should keep the way it is now for this PR. The existing `open func removeDataSet(_ dataSet: ChartDataSetProtocol!) -> Bool` uses `calcMinMax()`

Reply:
Maybe we can rethink that in another PR though.